src/components/Upload/UploadManager.tsx

```diff
@@ -0,0 +1,235 @@
+import {
+  Dialog,
+  DialogContent,
+  IconButton,
+  makeStyles,
+  DialogTitle as MuiDialogTitle,
+  Paper,
+  PaperProps,
+  Typography,
+} from '@material-ui/core';
+import {
+  Close as CloseIcon,
+  Maximize as MaximizeIcon,
+  Minimize as MinimizeIcon,
+} from '@material-ui/icons';
+import clsx from 'clsx';
+import React, {
+  FC,
+  memo,
+  useCallback,
+  useLayoutEffect,
+  useRef,
+  useState,
+} from 'react';
+import Draggable, { DraggableData, DraggableEvent } from 'react-draggable';
+import { useWindowSize } from 'react-use';
+import { useSession } from '../Session';
+import { useUploadManager } from './UploadContext';
+
+const PAPER_WIDTH = 480;
+
+const useStyles = makeStyles(({ palette, spacing }) => ({
+  root: {
+    pointerEvents: 'none',
+  },
+  paper: {
+    margin: 0,
+    pointerEvents: 'auto',
+    position: 'absolute',
+    top: 0,
+    left: 0,
+    width: `${PAPER_WIDTH}px`,
+  },
+  titleContainer: {
+    cursor: 'move',
+    display: 'flex',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    padding: spacing(1),
+    '&.collapsed': {
+      cursor: 'default',
+    },
+  },
+  title: {
+    cursor: 'move',
+    paddingLeft: spacing(1),
+    '&.collapsed': {
+      cursor: 'default',
+    },
+  },
+  titleButtons: {
+    marginLeft: 'auto',
+  },
+  titleButton: {},
+  contentContainer: {
+    padding: spacing(1),
+    '&.collapsed': {
+      display: 'none',
+    },
+  },
+  noUploadsText: {
+    color: palette.grey[400],
+  },
+}));
+
+const initialPosition = { x: null as null | number, y: null as null | number };
+
+const PaperComponent: FC<PaperProps & { isCollapsed: boolean }> = ({
+  ...props
+}) => {
+  const [position, setPosition] = useState({ x: 0, y: 0 });
+  const [isDragging, setIsDragging] = useState(false);
+  const paperRef = useRef<HTMLDivElement>(null);
+  const positionRef = useRef(initialPosition);
+  const { isCollapsed, ...rest } = props;
+
+  const { width: windowWidth, height: windowHeight } = useWindowSize();
+  const managerSize = {
+    width: (paperRef.current?.offsetWidth ?? 0) + 24,
+    height: (paperRef.current?.offsetHeight ?? 0) + 24,
+  };
+  const collapsedPosition = {
+    x: windowWidth - managerSize.width,
+    y: windowHeight - managerSize.height,
+  };
```

Comment:
Not sure where to put this but I don't think we need a separate position for the collapsed state. Having it collapse/expand where ever it seems fine. What do you think?

Reply:
I could go either way on it, so I went the easier way 🤣 If we let it stay where it is when collapsed, we have to do extra math + logic to keep track of its height. So I figured I'd let the design team tell me if they wanted that. Do you want to go ahead and do it, or keep this the way it is for now?

Reply:
Is that not happening already? The height is dynamic right?